### ISSUE TITLE
New option in TrailingCommentCheck

### ISSUE THREAD

**Issue by baratali** [checkstyle/checkstyle#1854]
In some places trailing comments are useful. For example:
```java
@Override
public int[] getAcceptableTokens() {
    return new int[] {
        TokenTypes.ASSIGN,            // '='
        TokenTypes.DIV_ASSIGN,        // "/="
        TokenTypes.PLUS_ASSIGN,       // "+="
        TokenTypes.MINUS_ASSIGN,      //"-="
        TokenTypes.STAR_ASSIGN,       // "*="
        TokenTypes.MOD_ASSIGN,        // "%="
        TokenTypes.SR_ASSIGN,         // ">>="
        TokenTypes.BSR_ASSIGN,        // ">>>="
        TokenTypes.SL_ASSIGN,         // "<<="
        TokenTypes.BXOR_ASSIGN,       // "^="
        TokenTypes.BOR_ASSIGN,        // "|="
        TokenTypes.BAND_ASSIGN,       // "&="
    };
}
```
Detailed description of the problem: http://roman-ivanov.blogspot.com/2013/10/trailing-comment-check-false-positives.html

Need to investigate cases when trailing comments are required and update TrailingCommentCheck to have an option which suppresses violations on them.

**Comment by wubbzee** [comment]
I'm on it!

**Comment by romani** [comment]
other tools demand trailing spaces to bypass violations https://errorprone.info/bugpattern/ListRemoveAmbiguous
